docs/source/addons/theme.md

```diff
@@ -0,0 +1,206 @@
+5. Now you can safely delete your project's `theme` folder, since the one in the add-on will take precedence.
+
+## Using your own escape hatch
+
+Volto theming uses SemanticUI theming capabilities to define and extend a theme for your site.
+However, while maintaining and playing well with the Semantic UI Volto base, using a traditional CSS approach can be done using the Less-based `extras` escape hatch.
+
+At the same time, one can entirely discard the extras escape hatch and add your own, by customizing the `theme.js` module in Volto.
+
+```js
+import 'semantic-ui-less/semantic.less';
+import '@plone/volto/../theme/themes/pastanaga/extras/extras.less';
+
+// You can add more entry points for theming
+import '@kitconcept/volto-light-theme/theme/main.scss';
+```
+
+While building your own escape hatch for theming, you can use the preprocessor of your choice (in the example, SCSS) while maintaining the "base" Volto theme, but customizing it using the resultant CSS.
+
+You can see an example of such a theme in: https://github.com/kitconcept/volto-light-theme
+
+## Modify a custom theme from another add-on
+
+Sometimes you have a custom theme that you want to reuse through all your projects, but with some differences, maintaining the base.
+Usually, the only option would be to use an add-on that adds more CSS to the base theme, using imports that will load after the theme.
+However, there is a problem with this approach.
+You cannot use existing theme variables, including breakpoints, on these new styles.
+Similarly, it gets somewhat detached from the normal flow of the loaded theme.
+The same applies for add-ons, as they are detached from the current theme.
+One could use a SemanticUI approach for making this work, but it's SemanticUI bound.
+
+```{warning}
+This is only possible when using your own escape hatch, and works only with SCSS-based themes, and not with SemanticUI themes, since it enables a couple of entry points that only support SCSS files.
+For an example of how it could be used, see: https://github.com/kitconcept/volto-light-theme
+```
+
+If your custom escape hatch defines a custom theme using SCSS, you can take advantage of this feature.
+Although not limited to this, it would be possible to extend this feature to add more entry points, using another preprocessor or theming approach.
+
+This feature enables two entry points: variables and main.
+From your add-on code, you can extend an existing theme by creating a file corresponding to each entry point:
+
+* `./src/theme/_variables.scss`
+* `./src/theme/_main.scss`
+
+### Variables (`addonsThemeCustomizationsVariables`)
+
+Use this entry point file to modify the original variables of the current loaded theme by adding the entry point after your own variable definitions in the theme.
+In the theme, it should be imported as shown below:
+
+```scss hl_lines="2"
+@import 'variables';
+@import 'addonsThemeCustomizationsVariables';
```

Comment:
I am not entirely sure I would suggest this as the default way. In most of my work I usually want customized variables applied before the theme variable, because theme variables in a proper scss theme **should** all have `!default`.

On the other hand, I understand that the change I propose would require also adding a few lines of examples to showcase the `!default` example and maybe a link to an explanation.

Reply:
Done.